### PR TITLE
Add make help

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this?**

/kind documentation
[skip ci]

**What does this PR do / why we need it**:

Add a `make help` command, extracting comments from Makefile (inspired by https://github.com/kubernetes/website/blob/master/Makefile)

**Which issue(s) this PR fixes**:

N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -94,9 +96,8 @@ sec:
 clean:
 	@rm -rf $(FILES)
 
-# install tools used for building, tests and  validations
 .PHONY: goget-tools
-goget-tools: goget-ginkgo
+goget-tools: goget-ginkgo ## install tools used for building, tests and  validations
```
Comment (dharmit) [inline Makefile:100]: nitpick - an extra space.
```suggestion
goget-tools: goget-ginkgo ## install tools used for building, tests and validations
```

File: Makefile [hunk 2]
```diff
@@ -159,241 +154,196 @@ test:
...
-test-integration-devfile:
+test-integration-devfile: ## Run devfile integration tests
 	ginkgo $(GINKGO_FLAGS) tests/integration/devfile/
 	ginkgo $(GINKGO_FLAGS_SERIAL) tests/integration/devfile/debug/
 
-# Run command's integration tests which are depend on service catalog enabled cluster.
 # Only service and link command tests are the part of this test run
 .PHONY: test-integration-service-catalog
-test-integration-service-catalog:
+test-integration-service-catalog: ## Run command's integration tests which are depend on service catalog enabled cluster.
```
Comment (dharmit) [inline Makefile:300]: Nitpick again.
```suggestion
test-integration-service-catalog: ## Run command's integration tests which are dependent on service catalog enabled cluster.
```

File: Makefile [hunk 3]
```diff
@@ -159,241 +154,196 @@ test:
...
 .PHONY: packages
-packages:
+packages: ## create deb and rpm packages using fpm in ./dist/pkgs/
 	./scripts/create-packages.sh
 
-# upload packages greated by 'make packages' to bintray repositories
 # run 'make cross' and 'make packages' before this!
 .PHONY: upload-packages
-upload-packages:
+upload-packages: ## upload packages greated by 'make packages' to bintray repositories
```
Comment (dharmit) [inline Makefile:334]:
```suggestion
upload-packages: ## upload packages created by 'make packages' to bintray repositories
```

**Review by dharmit — CHANGES_REQUESTED** [review]
Mostly nitpicks in my review.

**Comment by feloy** [comment]
Thanks @dharmit for the fixes

**Review by dharmit — APPROVED** [review]

**Pull request merged**